### ISSUE TITLE
azurerm >= 3.15.0: Error: parsing "/subscriptions/…/resourceGroups/…/providers/Microsoft.ServiceBus/namespaces/…/AuthorizationRules/…": parsing segment "staticAuthorizationRules": expected the segment "AuthorizationRules" to be "authorizationRules"

### ISSUE THREAD

**Issue by alexs77** [hashicorp/terraform-provider-azurerm#17720]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.15.0

### Affected Resource(s)/Data Source(s)

azurerm_servicebus_namespace_authorization_rule

### Terraform Configuration Files

```hcl
resource "azurerm_servicebus_namespace_authorization_rule" "…" {
  name         = local.name
  namespace_id = azurerm_servicebus_namespace.….id

  listen = true
  send   = true
}
```

### Debug Output/Panic Output

```shell
Error: parsing "/subscriptions/…/resourceGroups/…/providers/Microsoft.ServiceBus/namespaces/…/AuthorizationRules/…": parsing segment "staticAuthorizationRules": expected the segment "AuthorizationRules" to be "authorizationRules"

  with module.service_bus[0].azurerm_servicebus_namespace_authorization_rule.…,
  on modules/service_bus/service_bus.tf line 13, in resource "azurerm_servicebus_namespace_authorization_rule" "…":
  13: resource "azurerm_servicebus_namespace_authorization_rule" "…" {
```

### Expected Behaviour

No error. Same code works fine with AzureRM provider 3.14.0.

### Actual Behaviour

Doesn't work.

### Steps to Reproduce

…

**Comment by alexs77** [comment]
Just to make it clear: This is a regression/change from v3.14.0. It worked with 3.14.0.

**Comment by vhloba** [comment]
The same issue with v3.15.0
v3.14.0 works well.

**Comment by alexs77** [comment]
> The same issue with v3.15.0 v3.14.0 works well.

That's what I meant :)

- 3.15.0 -> broken
- <3.15.0 -> okay

**Comment by weldpua2008** [comment]
**If you can't upgrade azurerm (as I).**
You can use the following script to fix the state on "old" azurerm.
Please notice that some modules could require providing the subscription directly. So uncomment all related to `subscriptionFixApplied` and using sed fix the relevant files.


```bash
subscriptionFixApplied=""
LOG_PREFIX="${HOME}/.$0"
STATE_LIST_PATH="${LOG_PREFIX}/azurerm_eventhub_authorization_rule_$(date "+%Y%m%d_%H%M").txt"
mkdir -p ${LOG_PREFIX}
for st in $(terraform state list|grep azurerm_eventhub_authorization_rule| tee ${STATE_LIST_PATH});do
    resourceName="$(echo $st| tr -d '\r')"
    [[ -z "${resourceName}" ]] && continue
    [[  "${resourceName}" = " " ]] && continue
    resourceLogPath="${LOG_PREFIX}/${resourceName}$(date "+%Y%m%d_%H%M").txt"
    resourceLogCMDPath="${LOG_PREFIX}/${resourceName}_cmd$(date "+%Y%m%d_%H%M").txt"
    resourceId=$(terraform state  show $resourceName| tee ${resourceLogPath}  |grep id| grep --color=never -Eo '".*"'|cut -d '"' -f2)
    [[ -z "${resourceId}" ]] && continue
    [[  "${resourceId}" = " " ]] && continue
    [[ "${resourceId}" != *authorizationrules* ]] && continue
    resourceIdFixed=$(echo $resourceId| sed 's,authorizationrules,authorizationRules,' )
    echo " => ${resourceId}"
    echo "terraform state rm ${resourceName}" | tee ${resourceLogCMDPath}
    echo "terraform import  \"${resourceName}\" \"${resourceIdFixed}\"" | tee -a ${resourceLogCMDPath}
    # subscriptionId=$(echo $resourceIdFixed| grep -Eo 'subscriptions/(.*)/resourcegroups'| grep -Eo '/(.*)/'| head -1 | tr -d '/')
    # Fixing
    # Error: Error building AzureRM Client: 1 error occurred:
    #    * A Subscription ID must be configured when authenticating as a Service Principal using a Client Secret.
    # if [[ "${subscriptionFixApplied}" != "yes" ]];then
    #     if [ "$(uname)" == "Darwin" ]; then
    #         macOssedFlags=" ''"
    #     fi
    #     if [[ -e "${subscriptionIdFixPath}" ]];then
    #         cp "${subscriptionIdFixPath}" "${subscriptionIdFixPath}__"
    #         sed  -i ${macOssedFlags} 's/subscription_id.*$/subscription_id = "'${subscriptionId}'"/' "${subscriptionIdFixPath}"
    #     fi
    #     subscriptionFixApplied="yes"
    # fi   
    terraform state rm "${resourceName}"
    terraform import  "${resourceName}" "${resourceIdFixed}" || terraform import  "${resourceName}" "${resourceIdFixed}"
done
# if [[ -e "${subscriptionIdFixPath}__" ]];then
#     cp "${subscriptionIdFixPath}__" "${subscriptionIdFixPath}"
#     rm -f "${subscriptionIdFixPath}__"
# fi

**Comment by enorlando** [comment]
We are also encountering the same issue when using v3.15.0 of the provider

**Comment by alexs77** [comment]
Still present with v3.15.1 of the provider.

**Comment by bigglesuk69** [comment]
There was a similar issue with another resource a while back (https://github.com/hashicorp/terraform-provider-azurerm/issues/15549) and it was caused when the SDK was changed. I know the SDK on the Servicebus resources was changed recently so could be related.

**Comment by Marcus-James-Adams** [comment]
> Error: parsing "/subscriptions/..../resourceGroups/..../providers/Microsoft.ServiceBus/namespaces/.../AuthorizationRules/readwrite": parsing segment "staticAuthorizationRules": expected the segment "AuthorizationRules" to be "authorizationRules"
> │
> │   with azurerm_servicebus_namespace_authorization_rule.msgprocs-readwrite,
> │   on servicebus.tf line 26, in resource "azurerm_servicebus_namespace_authorization_rule" "msgprocs-readwrite":
> │   26: resource "azurerm_servicebus_namespace_authorization_rule" "msgprocs-readwrite" {
> │
> ╵
> 
> C:\repo\cloud_infrastructure\projects\platform\c17\UK_South\3.7.msgprocs>terraform version
> Terraform v1.2.1
> on windows_amd64
> + provider registry.terraform.io/cloudflare/cloudflare v3.20.0
> + provider registry.terraform.io/hashicorp/azuread v2.26.1
> + provider registry.terraform.io/hashicorp/azurerm v3.16.0


Still occurring on 3.16.0

**Comment by nomaster** [comment]
Also still occurring on 3.17.0

**Comment by Vercidium** [comment]
Hi all,

Is there a workaround available for this? I attempted to change the version specified in `main.tf` from `>= 2.26.0` to `=3.14.0` like so:

```
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "=3.14.0"
    }
  }
}
```

However my `terraform init` task in my Azure pipeline still says it's searching for `>= 2.26.0` and installs the latest version:

```
Initializing provider plugins...
Finding hashicorp/azurerm versions matching ">= 2.26.0"...
Installing hashicorp/azurerm v3.17.0...
Installed hashicorp/azurerm v3.17.0 (self-signed, key ID [0m[1m34365D9472D7468F[0m[0m)
```

**Comment by alexs77** [comment]
> However my `terraform init` task in my Azure pipeline still says it's searching for `>= 2.26.0` and installs the latest version:

Try `terraform init -upgrade`. Despite the name being "upgrade", `terraform init` will then also happily do a downgrade, when you specify `-upgrade`.

**Comment by wiebeck** [comment]
This seems to be fixed with provider version 3.19.0 finally. The changelog doesn't mention a bigfix for this but I assume the update of the Azure SDK fixed it.

Hooray!

**Comment by tombuildsstuff** [comment]
Fixed in [v3.19.0](https://github.com/hashicorp/terraform-provider-azurerm/milestone/194)
